packages/tools/devtools/devtools-core/src/data-visualization/SharedTreeVisualizer.ts

```diff
@@ -0,0 +1,245 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import type {
+	FieldMapObject,
+	JsonableTree,
+	SharedTreeContentSnapshot,
+	TreeNodeStoredSchema,
+	TreeTypeSet,
+} from "@fluidframework/tree/internal";
+import {
+	EmptyKey,
+	LeafNodeStoredSchema,
+	MapNodeStoredSchema,
+	ObjectNodeStoredSchema,
+} from "@fluidframework/tree/internal";
+import type { SharedTreeLeafNode, VisualSharedTreeNode } from "./VisualSharedTreeTypes.js";
+import { VisualSharedTreeNodeKind } from "./VisualSharedTreeTypes.js";
+import {
+	type VisualChildNode,
+	VisualNodeKind,
+	type VisualValueNode,
+	type VisualTreeNode,
+} from "./VisualTree.js";
+
+/**
+ * Converts the output of {@link sharedTreeVisualizer} to {@link VisualChildNode} type containing `schema` and `children` fields.
+ */
+export function toVisualTree(tree: VisualSharedTreeNode): VisualValueNode | VisualTreeNode {
+	if (tree.kind === VisualSharedTreeNodeKind.LeafNode) {
+		const result: VisualValueNode = {
+			value: tree.value,
+			nodeKind: VisualNodeKind.ValueNode,
+			tooltipContents: tree.schema.schemaName,
+		};
+		return result;
+	} else {
+		const children: Record<string, VisualChildNode> = {};
+
+		for (const [key, value] of Object.entries(tree.fields)) {
+			const child = toVisualTree(value);
+			children[key] = child;
+		}
+
+		return {
+			children,
+			nodeKind: VisualNodeKind.TreeNode,
+			tooltipContents: tree.schema.allowedTypes,
```

Comment:
Short term hack: create a helper function to generate a nicely formatted string that looks like:
```
schemaName: <schema.name>
allowedTypes: 
    * foo: <foo>
    * bar: <bar>
```

Reply:
Totally fine for this to be in a follow up PR :)